### PR TITLE
Improve Windows support

### PR DESCRIPTION
**Pull request by Schroedingers-Cat**
This PR improves the windows support by the following changes:
* Add READM.md explaining how to build the clap host on Windows
* Integrate vcpkg with CMake so dependencies will be managed automatically
* make some casts explicit so the code will compile with MSVC

There's still an issue left when starting the built example host. The only solution I found until now is to manually build Qt6 from source (https://github.com/Schroedingers-Cat/clap/tree/feature/build-host-on-windows#windows). However, this PR's scope is not to change how certain dependencies are handled, so I'm leaving that one out.  
Also, the recent addition of the porttime library to the linker is causing problems. ATM, I'm manually removing that because it doesn't seem to be necessary.

### CODE REVIEW

**Comment by Schroedingers-Cat** [comment]
I added a small change to this PR: the vcpkg toolchain file will now be added only on Windows.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 3.17)
 cmake_policy(SET CMP0100 NEW)  # handle .hh files
+set(CMAKE_TOOLCHAIN_FILE "./vcpkg/scripts/buildsystems/vcpkg.cmake")
```
[the remark this answers is not in the document]
Comment (Schroedingers-Cat) [reply CMakeLists.txt]: My bad, I overlooked that one.  
e37056cc037161a142d04e3d7868f5e7566bdfd7 removes that, aa66fe1392604181bdb4646a046744fb1bbaeb7f adds a CMake preset for VS+vcpkg and 2f41f93612cefdefa8cb7e16b04b0c1d1d28caec updates this in the README.md

File: host/CMakeLists.txt [hunk 2]
```diff
@@ -1,12 +1,18 @@
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
 
+if (APPLE OR UNIX)
```
[the remark this answers is not in the document]
Comment (Schroedingers-Cat) [reply host/CMakeLists.txt]: You are right (dd580a6c89deac5a95a1e18e07a2e4131d7ae815)

File: vcpkg.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {
-  "name": "main",
-  "version-string": "latest",
+  "name": "clap-host",
```
[the remark this answers is not in the document]
Comment (abique) [reply vcpkg.json:2]: My misunderstanding apparently.

**Pull request merged**